pulser/simulation/simulation.py

```diff
@@ -333,6 +334,11 @@ def add_noise(self, noise_type):
             self._prepare_spam_detune()
         if noise_type == 'doppler':
             self.init_doppler_sigma()
+        if noise_type == 'dephasing':
+            if self.basis_name == 'digital':
+                raise ValueError("Cannot include dephasing noise in digital "
+                                 "basis.")
+            self.init_dephasing()
```

Comment:
We have to add the case where `self.basis_name == 'all'` or else the operators don't have the right dimension.